apps/admin-ui/src/spa/reservations/[id]/ReservationTitleSection.tsx

```diff
@@ -134,6 +126,27 @@ const ReservationTitleSection = forwardRef<HTMLDivElement, Props>(
       }
     })(order?.status);
 
+    const stateLabelProps = ((
+      s?: Maybe<ReservationStateChoice>
+    ): { type: StatusLabelType; icon: JSX.Element } => {
+      switch (s) {
+        case ReservationStateChoice.Created:
+          return { type: "draft", icon: <IconPen aria-hidden /> };
+        case ReservationStateChoice.WaitingForPayment:
+          return { type: "alert", icon: <IconEuroSign aria-hidden /> };
+        case ReservationStateChoice.RequiresHandling:
+          return { type: "info", icon: <IconCogwheel aria-hidden /> };
+        case ReservationStateChoice.Confirmed:
+          return { type: "success", icon: <IconCheck aria-hidden /> };
+        case ReservationStateChoice.Denied:
+          return { type: "error", icon: <IconCross aria-hidden /> };
+        case ReservationStateChoice.Cancelled:
+          return { type: "neutral", icon: <IconCross aria-hidden /> };
+        default:
+          return { type: "neutral", icon: <IconQuestionCircle aria-hidden /> };
+      }
+    })(reservation?.state);
```

Comment:
I'd make this into a free function (out of the component) not anonymous with an instant call.

Also `aria-hidden="true"`